### PR TITLE
Update release/1.3.1 (starting from release/1.3.0)

### LINKED ISSUE #516: add eckit 1.22.1 for release v5
eckit 1.22.1 contains features that would be useful in jedi oops (https://github.com/JCSDA-internal/oops/issues/2140).
Would be helpful if it were added to stack and made a default for jedi stacks in the next release (v5).

### LINKED ISSUE #518: Add version 1.1.2 of gsibec
**Please describe the package or library you would like to add to spack-stack.**

`gsibec@1.1.2` is now required by `fv3-jedi` (in fact, it needs at least 1.1.1, but there is a bug in 1.1.1 that we fixed in 1.1.2)

**What applications will be using this package or library?**
JEDI

**Is there already a package or library in spack-stack that provides this, or related, functionality?**
`gsibec` is already in spack-stack

**Additional context**
n/a

**Will This Package be Needed in a NOAA Operational Application?**
n/a (at least not for now)

### LINKED ISSUE #542: Update odc from 1.4.5 to 1.4.6
**Is your feature request related to a problem? Please describe.**
We should update odc to 1.4.6, which includes a number of bug fixes and other minor updates (no major changes).

https://github.com/ecmwf/odc/releases/tag/1.4.6

**Describe the solution you'd like**
See above

**Additional context**
n/a

### LINKED ISSUE #544: add CRTM v2.4.1 for JEDI/UFO applications
**Please describe the package or library you would like to add to spack-stack.**
This is a request to add CRTM v2.4.1 from https://github.com/JCSDA/crtm  `release/crtm_jedi_v2.4.1` branch, which is tagged at 
[v2.4.1-jedi](https://github.com/JCSDA/crtm/tree/v2.4.1-jedi).   Unless there's a need identified via this PR., the develop branch should not change.   If necessary, I will issue a new tag if the develop branch does change.

An associated binary file tarball is located at: 
https://ftp.ssec.wisc.edu/pub/s4/CRTM/file/crtm_coefficients_2.4.1_skylab_4.0.tar.gz

This will be transitioned to UCAR gdex once I've got enough testing / feedback from users.

**What applications will be using this package or library?**
JCSDA UFO + downstream applications

**Is there already a package or library in spack-stack that provides this, or related, functionality?**
Prior CRTM packages

**Will This Package be Needed in a NOAA Operational Application?**
No, this is not recommended.  In fact, it should not be used operationally, since we'll be transitioning all capabilities in v2.4.1 into v3.0.0, and no further development will occur in the v2.x versions of CRTM.

CRTM is public domain code.

### LINKED ISSUE #546: Create internal bug fix release spack-stack-1.3.1
**Is your feature request related to a problem? Please describe.**

We've run into problems with Skylab when using spack-stack-1.3.0 and we've tracked it down to hdf5@1.14.0. With 1.12.2, the issues go away (masked numpy array operations directly on an hdf5 object being very, very slow - one subtraction taking several minutes when it should take a fraction of a second).

**Describe the solution you'd like**

JCSDA would therefore like to create a spack-stack-1.3.1 (internal) bug fix release as early as next week, to be installed only on selected platforms. In addition to reverting the hdf5 upgrade, we’d like to add a few new versions of eckit, atlas etc.

As discussed in the spack-stack slack channel, I’ll create the 1.3.1 bug fix release on top of the 1.3.0 release branches, install it side by side to the existing 1.3.0 releases on Orion, S4, Discover, Cheyenne (and AWS Parallel Cluster). EMC and EPIC can continue using 1.3.0 w/o restrictions, or try 1.3.1 with the newer ESMF library if that is of interest (to check if ESMF@8.4.1 really requires hdf5@1.13.0)? I’ll also remove the debug builds of ESMF and MAPL in that version.

**Additional context**

Work on develop can continue as normal, we’ll just merge any applicable changes from the 1.3.1 tags back to develop later.

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

Update the default versions for crtm (for JEDI), ecbuild, eckit, fckit, ecmwf-atlas, gsibec, odc. Most importantly, downgrade hdf5 from 1.14.0 to 1.12.2 and py-h5py from 3.7.0 to 3.6.0 to fix incredibly slow Python scripts that perform masked array operations on hdf5 objects (c.f. https://github.com/JCSDA-internal/ewok/issues/623 if you have permissiosn to see this).

Update documentation for spack-stack-1.3.1 installations, where available, and update the container `README.md` file for building a ufs-weather-model container (this is still work in progress, but needs to be documented somewhere).

Also: remove unused `skylab-no-python` template and remove `ufs-weather-model-env+debug` from several templates. Add missing `jedi-mpas-env` virtual package to `unified-dev` (does not install additional packages, just the convenience module to load whatever is needed for jedi-mpas).

## Installations

spack-stack-1.3.1 installed side by side with 1.3.0 on the following systems:
- [x] Orion (Intel, GNU)
    - [x] spack-stack-1.3.1 installation
    - [x] addition of crtm@v2.4-jedi.2, crtm@v3.0.0-rc.1, esmf@8.4.1, mapl@2.35.2
    - [x] addition of missing jedi-mpas-env@unified-dev
- [x] S4 (Intel, GNU)
    - [x] spack-stack-1.3.1 installation
    - [x] addition of crtm@v2.4-jedi.2, crtm@v3.0.0-rc.1, esmf@8.4.1, mapl@2.35.2
    - [x] addition of missing jedi-mpas-env@unified-dev
- [x] Discover (Intel, GNU)
    - [x] spack-stack-1.3.1 installation
    - [x] addition of crtm@v2.4-jedi.2, crtm@v3.0.0-rc.1, esmf@8.4.1, mapl@2.35.2
    - [x] addition of missing jedi-mpas-env@unified-dev
- [x] Cheyenne (Intel, GNU)
    - [x] spack-stack-1.3.1 installation
    - [x] addition of crtm@v2.4-jedi.2, crtm@v3.0.0-rc.1, esmf@8.4.1, mapl@2.35.2
    - [x] addition of missing jedi-mpas-env@unified-dev
- [x] JCSDA AWS Parallel Cluster Ubuntu (Intel, GNU)
    - [x] spack-stack-1.3.1 installation
    - [x] addition of crtm@v2.4-jedi.2, crtm@v3.0.0-rc.1, esmf@8.4.1, mapl@2.35.2
    - [x] addition of missing jedi-mpas-env@unified-dev
- [x] JCSDA AWS Red Hat 8 AMI (GNU)
    - [x] spack-stack-1.3.1 installation
    - [x] addition of crtm@v2.4-jedi.2, crtm@v3.0.0-rc.1, esmf@8.4.1, mapl@2.35.2
    - [x] addition of missing jedi-mpas-env@unified-dev
    - [x] creation of new AMI, public, in us-east-1 and us-east-2

## Issues

- fixes #546 
- fixes #544 
- fixes #542 
- fixes #518 
- fixes #516 

## Dependencies

Waiting on https://github.com/NOAA-EMC/spack/pull/259 (merged 04/20/2023)

### CODE REVIEW

File: configs/containers/README.md [hunk 1]
```diff
@@ -25,9 +25,36 @@ To avoid hardcoding specs in the generic container recipes, we keep the specs li
     # py-mysql-connector-python@8.0.32, 
 ```
 
+### spack-stack-1.3.1 / ufs-weather-model-x.y.z containers for ufs-weather-model as of April 18, 2023
```
Comment (climbfuji) [inline configs/containers/README.md:28]: Note. This is still work in progress

File: doc/source/PreConfiguredSites.rst [hunk 2]
```diff
@@ -464,11 +571,36 @@ For ``spack-stack-1.3.0`` with Intel, run the following commands/load the follow
...
+
+For ``spack-stack-1.3.1`` with GNU, run the following commands/load the following modules:
+
+   module purge
+   ulimit -s unlimited
+   module use /mnt/experiments-efs/skylab-v4/spack-stack-1.3.1/envs/unified-env/install/modulefiles/Core
+   module load stack-gcc/9.4.0
+   module load stack-openmpi/4.1.4
+   module load stack-python/3.10.8
+   module available
```
Comment (BenjaminTJohnson) [inline doc/source/PreConfiguredSites.rst:595]: The only addition I would suggest (not sure that it goes here), is that I have to load one of the env modules, such as `module load jedi-fv3-env/unified-dev`  (at least on S4 this was necessary).

**Review by BenjaminTJohnson — APPROVED** [review]

File: doc/source/PreConfiguredSites.rst (hunk 2 above)
Comment (climbfuji) [reply doc/source/PreConfiguredSites.rst:595]: Correct, that is part of jedi-docs. spack-stack documentation needs to be independent of the application (ufs-weather-model, global-workflow, skylab, ...) therefore we didn't add those module load commands here.

**Review by shlyaeva — APPROVED** [review]
Changes look good to me, as I expected!

**Pull request merged**